src/library/browse/browsethread.cpp

```diff
@@ -144,15 +171,19 @@ void BrowseThread::populateModel() {
     int row = 0;
     // Iterate over the files
     while (fileIt.hasNext()) {
-        // If a user quickly jumps through the folders
-        // the current task becomes "dirty"
-        m_path_mutex.lock();
-        auto newPath = m_path;
-        m_path_mutex.unlock();
-
-        if (thisPath.info() != newPath.info()) {
-            qDebug() << "Abort populateModel()";
-            populateModel();
```

Comment:
I didn't like this recursion, better give the control back to the run() method to handle the new request.